### DISCUSSION TITLE
Allow non-existant field to pass resource form fill on saving

### DISCUSSION THREAD

**Discussion by LuanHimmlisch** [filamentphp/filament#6871] [Help]
I have a field on a resource that **doesn't** correspond to a column of the model.

```php
Forms\Components\Select::make('taxonomy')
    ->options(TaxonomyTerm::all()->pluck('name', 'slug'))
```

I need this field to pass the saving from Filament, since I use this offset on the boot for custom logic:

```php
protected static function boot()
{
    static::saving(function (self $model) {
        if (!$model->offsetExists('taxonomy')) {
            return;
        }
        /* Custom logic... */
    });
}
```

I can't find a way to prevent Filament from correctly detecting the field is not a column and erasing it from the data.

**Comment by danharrin** [comment]
Its not Filament, it's Eloquent that does that if its not `$fillable`. If you want to access it, you'd need to do so on the Filament form side in a lifecycle hook or similar?

**Comment by Patabugen** [comment]
You haven't described what the issue you're having, so this might not be helpful - but since I came across an unanswered question I thought I'd post something that might be useful.

You can modify the data before it gets passed to Eloquent with this method on your page: 

```php
    protected function mutateFormDataBeforeSave(array $data): array
    {
        unset($data['taxonomy']);
        // Maybe do your custom logic here too
        return $data;
    }
```

https://filamentphp.com/docs/4.x/resources/editing-records#customizing-data-before-saving
